src/overlays/ovl_i1/sf_tr360.c

```diff
@@ -1,9 +1,233 @@
+    sinRotY = __sinf(actor->obj.rot.y * M_DTOR);
+    cosRotY = __cosf(actor->obj.rot.y * M_DTOR);
+    sp54.z = actor->fwork[4] - actor->obj.pos.x;
+    sp54.y = actor->fwork[5] - actor->obj.pos.y;
+    sp54.x = actor->fwork[6] - actor->obj.pos.z;
+    if (!((actor->index + gGameFrameCount) & 7)) {
+        actor->fwork[19] = Math_RadToDeg(Math_Atan2F(sp54.z, sp54.x));
+        actor->fwork[20] = Math_RadToDeg(Math_Atan2F(sp54.y, sqrtf(SQ(sp54.z) + SQ(sp54.x))));
+    }
+    var_fv0 = actor->fwork[20];
+    temp_v0 = func_i1_80198DCC(actor, sinRotY, cosRotY);
+    if (temp_v0 != 0) {
+        var_fv0 += 40.0f * (f32) temp_v0;
+        if (var_fv0 >= 360.0f) {
+            var_fv0 -= 360.0f;
+        }
+        if (var_fv0 < 0.0f) {
+            var_fv0 += 360.0f;
+        }
+    } else if ((actor->obj.pos.y < (gGroundLevel + 50.0f)) && (var_fv0 > 180.0f)) {
+        var_fv0 = 0.0f;
+        actor->unk_0F4.x = 0.0f;
+    }
+    Math_SmoothStepToAngle(&actor->unk_0F4.x, var_fv0, 0.5f, 1.0f, 0.0001f);
+    temp_fv1 = Math_SmoothStepToAngle(&actor->unk_0F4.y, actor->fwork[19], 0.5f, 1.0f, 0.0001f) * 30.0f;
+    if (temp_fv1 < 0.0f) {
+        var_fv0_2 = temp_fv1 * -1.0f;
+    } else {
+        var_fv0_2 = 360.0f - temp_fv1;
+    }
+    Math_SmoothStepToAngle(&actor->obj.rot.z, var_fv0_2, 0.1f, 3.0f, 0.01f);
+    actor->obj.rot.x = -actor->unk_0F4.x;
+    actor->obj.rot.y = actor->unk_0F4.y;
+    actor->vel.x = actor->fwork[13] + (sinRotY * (cosRotX * 35.0f));
+    actor->vel.y = actor->fwork[14] + (-sinRotX * 35.0f);
+    actor->vel.z = actor->fwork[12] + (cosRotY * (cosRotX * 35.0f));
+    actor->fwork[13] -= (actor->fwork[13] * 0.1f);
+    actor->fwork[14] -= (actor->fwork[14] * 0.1f);
+    actor->fwork[12] -= (actor->fwork[12] * 0.1f);
+    if ((actor->obj.pos.y < gGroundLevel + 40.0f) && (actor->vel.y < 0.0f)) {
+        actor->obj.pos.y = gGroundLevel + 40.0f;
+        actor->vel.y = 0.0f;
+    }
+    func_8003088C(actor);
+    unkEntity28 = &gUnkEntities28[actor->index];
+    unkEntity28->unk_00 = 1;
+    unkEntity28->unk_02 = actor->unk_0E4;
+    unkEntity28->pos.x = actor->obj.pos.x;
+    unkEntity28->pos.y = actor->obj.pos.y;
+    unkEntity28->pos.z = actor->obj.pos.z;
+    unkEntity28->unk_10 = actor->unk_0F4.y + 180.0f;
+    if (actor->iwork[8] != 0) {
+        actor->iwork[8]--;
+    }
+}
+
+typedef struct {
+    u16* msg;
+    u16 unk;
+} UnkMsg_D_i1_8019AE50;
+
+UnkMsg_D_i1_8019AE50 D_i1_8019AE50[] = {
+    { gMsg_ID_23028, 224 }, { gMsg_ID_23029, 160 }, { gMsg_ID_23030, 224 }, { gMsg_ID_23031, 256 },
+    { gMsg_ID_23016, 160 }, { gMsg_ID_23024, 160 }, { gMsg_ID_23019, 256 }, { gMsg_ID_23015, 160 },
+    { gMsg_ID_23017, 160 }, { gMsg_ID_23025, 160 }, { gMsg_ID_23002, 160 }, { gMsg_ID_23004, 160 },
+    { gMsg_ID_23005, 160 }, { gMsg_ID_23023, 160 }, { gMsg_ID_23006, 160 }, { gMsg_ID_23007, 256 },
+    { gMsg_ID_20330, 0 },   { gMsg_ID_23026, 256 }, { gMsg_ID_23032, 256 }, { NULL, 0xFFFF },
```

Comment:
```suggestion
    { gMsg_ID_20330, 0 },   { gMsg_ID_23026, 256 }, { gMsg_ID_23032, 256 }, { NULL, -1 },
```
This looks better to me, even if it is technically an unsigned value.